src/bvals/cc/sparse_bvals_cc_in_one.cpp

```diff
@@ -0,0 +1,393 @@
+  
+  Mesh* pmesh = md->GetMeshPointer(); 
+
+  // Allocate channels sending from active data and then check to see if 
+  // if buffers have changed
+  bool rebuild = false;
+  int nbound = 0; 
+  {
+  WriteRegion region("allocate send resource");
+  IterateBoundaries(md, [&](sp_mb_t pmb, sp_mbd_t rc, nb_t &nb, const sp_cv_t v) {
+    PARTHENON_DEBUG_REQUIRE(pmesh->boundary_comm_map.count({pmb->gid, nb.snb.gid, v->label()}) > 0, 
+        "Boundary communicator does not exist");
+    auto& buf = pmesh->boundary_comm_map[{pmb->gid, nb.snb.gid, v->label()}];
+    
+    if (v->IsAllocated()) {
+      buf.Allocate();
+    } else { 
+      buf.Free();
+    }
+
+    if (nbound < md->send_bnd_info_h.size()) {    
+      rebuild = rebuild || !UsingSameResource(md->send_bnd_info_h(nbound).buf, buf);
+    } else { 
+      rebuild = true; 
+    }
+
+    ++nbound;
+  });
+  } 
+
+  if (rebuild) {
+    WriteRegion region("rebuild send_bnd_info");
+    md->send_bnd_info = BufferCache_t("send_boundary_info", nbound);
+    md->send_bnd_info_h = Kokkos::create_mirror_view(md->send_bnd_info);
+
+    int iarr = 0;
+    IterateBoundaries(md, [&](sp_mb_t pmb, sp_mbd_t rc, nb_t &nb, const sp_cv_t v) {
+      auto& buf = pmesh->boundary_comm_map[{pmb->gid, nb.snb.gid, v->label()}];
+      
+      md->send_bnd_info_h(iarr) = BndInfo::Sender(pmb, nb, v); 
+      md->send_bnd_info_h(iarr).buf = buf; 
+      md->send_bnd_info_h(iarr).allocated = v->IsAllocated();
+       
+      ++iarr;
+    });
+    Kokkos::deep_copy(md->send_bnd_info, md->send_bnd_info_h); 
+  }   
+  
+  // Restrict 
+  {
+    WriteRegion region("restrict");
+    auto &rc = md->GetBlockData(0);
+    auto pmb = rc->GetBlockPointer();
+    IndexShape cellbounds = pmb->cellbounds;
+    IndexShape c_cellbounds = pmb->c_cellbounds;
+    cell_centered_refinement::Restrict(md->send_bnd_info, cellbounds, c_cellbounds);
+  }
+
+  // Load buffer data 
+  const Real threshold = Globals::sparse_config.allocation_threshold;
+  auto& bnd_info = md->send_bnd_info; 
+  
+  ParArray1D<bool> sending_nonzero_flags("sending_nonzero_flags", nbound);
+  auto sending_nonzero_flags_h = Kokkos::create_mirror_view(sending_nonzero_flags);
+  {
+  WriteRegion region("launch buffer load");
+  Kokkos::parallel_for(
+    "SendBoundaryBuffers",
+    Kokkos::TeamPolicy<>(parthenon::DevExecSpace(), nbound, Kokkos::AUTO),
+    KOKKOS_LAMBDA(parthenon::team_mbr_t team_member) {
+      const int b = team_member.league_rank();
+      
+      sending_nonzero_flags(b) = false;
+      if (!bnd_info(b).allocated) return;
+
+      const int &si = bnd_info(b).si;
+      const int &ei = bnd_info(b).ei;
+      const int &sj = bnd_info(b).sj;
+      const int &ej = bnd_info(b).ej;
+      const int &sk = bnd_info(b).sk;
+      const int &ek = bnd_info(b).ek;
+      const int Ni = ei + 1 - si;
+      const int Nj = ej + 1 - sj;
+      const int Nk = ek + 1 - sk;
+      const int &Nv = bnd_info(b).Nv;
```

Comment:
Another place where `Nt` and `Nv` should be included, along with updated indices inside the `parallel_for` below